sklearn/utils/seq_dataset.pxd

```diff
@@ -7,6 +7,7 @@ cimport numpy as np
 
 cdef class SequentialDataset:
     cdef int current_index
```

Comment:
There are three things called index here. I hope they are documented somewhere.

Also, I'm not sure if we should consider this public interface that's problematic to change... Can't we use `index_data_ptr[current_index]` directly?

Reply:
Right, I simplified the call using simply `index_data_ptr[current_index]`.
I also added a bit of documentation there.